src/ray/gcs/store_client/test/redis_store_client_test.cc

```diff
@@ -14,6 +14,7 @@
 
 #include "ray/gcs/store_client/redis_store_client.h"
 
+#include <boost/optional/optional_io.hpp>
```

Comment:
this is required to build..

previously we were not even using the `auto_http_archive` 's import for googletest. the package was dragged in by some other package.